src/libraries/Pool.sol

```diff
@@ -237,18 +237,12 @@ library Pool {
         }
     }
 
-    // the top level state of the swap, the results of which are recorded in storage at the end
-    struct SwapState {
-        // the amount remaining to be swapped in/out of the input/output asset
-        int256 amountSpecifiedRemaining;
-        // the amount already swapped out/in of the output/input asset
-        int256 amountCalculated;
-        // current sqrt(price)
+    // the results after a swap, written to storage and returned
+    struct SwapResult {
+        // the final current sqrt(price)
         uint160 sqrtPriceX96;
         // the tick associated with the current price
         int24 tick;
-        // the global fee growth of the input token
-        uint256 feeGrowthGlobalX128;
```

Comment:
love this simplification of only returning the values needed in the pool manager